src/mender-auth/cli/actions.cpp

```diff
@@ -143,6 +143,14 @@ error::Error DaemonAction::Execute(context::MenderContext &main_context) {
 
 	events::EventLoop loop {};
 
+	events::SignalHandler signal_handler {loop};
+
+	signal_handler.RegisterHandler(
+		{SIGTERM, SIGINT, SIGQUIT}, [&loop](events::SignalNumber signum) {
+			log::Info("Termination signal received, shutting down gracefully");
+			loop.Stop();
+		});
```

Comment:
We should check the return value
```suggestion
	err = signal_handler.RegisterHandler(
		{SIGTERM, SIGINT, SIGQUIT}, [&loop](events::SignalNumber signum) {
			log::Info("Termination signal received, shutting down gracefully");
			loop.Stop();
		});
	if (err != error::NoError) {
		return err;
	}
```